Developer Documentation Updates

jmdegeorge opened the pull request:
Updating wrims developer documentation with latest gradle devops support.

jmdegeorge committed:
Updating developer documentation for building and debugging WRIMS from a new workspace.

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+- Gradle 8.5
+- Git
+- IDE (IntelliJ, Eclipse, etc.)
+- <USER-DIR>\.gradle\Gradle.properties configured with token for access to the CentralValleyModeling GitHub repository
```
The `<USER-DIR>` part of this string is lost in markdown. May need to escape the `<>` for this to render properly.

jmdegeorge replied:
Resolved

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+
+## 2. Override the build the version number
+The version number is configured to default to the branch name to support automatic tagged version
+via the github actions build system. 
```
It may be useful to include a direct link to the releases page to choose a valid tagged version.

jmdegeorge replied:
This doesn't need to match a valid tagged version.

jmdegeorge replied:
I added a note with additional examples.

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+The installer can be built by running the "zipWrimsGui" task in the wrims-gui module.
+
+The generated wrims install zip file will be located in the /wrims-gui/build/installer
+folder. The installer file will be named: wrims_gui_x64_<version>.zip
```
The `<version>` string is also not correctly rendered. See above.

zack-rma replied:
There are other locations where this is the case, but I'll stop pointing them out.

jmdegeorge replied:
Resolved

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+
+## 5. Run the WRIMS GUI Application
+
+Once the installer has been built, the WRIMS gui application can launched by running one of the following files
```
Not a big deal, but it would be nice to keep the capitalization of GUI consistent.

jmdegeorge replied:
Fixed

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+## 5. Run the WRIMS GUI Application
+
+Once the installer has been built, the WRIMS gui application can launched by running one of the following files
+from the installer directory (wirms-gui\build\isntaller\wrims_gui_x64_<version>):
```
Typo in `installer` in path.

jmdegeorge replied:
Fixed

zack-rma commented on README.build-gradle.md:
````diff
@@ -0,0 +1,146 @@
+You can build the installer and patch by running the following linux command from the root of the project:
+```
+./gradlew :wrims-gui:zipWrimsGui
+./gradlew :wrims-gui:zipWrimsPatch
````
I see building the patch is mentioned here, but instructions/details are further below. Is building the patch a necessary part of zipping the GUI? If it is, then maybe this can be a bit clearer.

jmdegeorge replied:
Moved the patch command lines to the patch block

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+The generated wrims patch zip file will be located in the /wrims-gui/build/patch
+folder. The patch file will be named: wrims_patch_v2.2.0_basis.zip
+
+NOTE: The patch zip version is currently hard-coded to "v2.2.0_basis" in the build.gradle file for
```
I thought [this](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) might be helpful to improve the readability of notes and warnings.

jmdegeorge replied:
I applied that formatting to the NOTE elements but that markup only applies when viewed in GitHub and doesn't show up as nice in IntelliJ (for future reference).

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+
+## How to build wrims installer From GitHub Releases
+The installer and patch zip files are automatically generated when a new "Release" is created from
+the github wrims site (https://github.com/CentralValleyModeling/wrims).
```
Minor note: capitalization of WRIMS and GitHub should also be consistent.

jmdegeorge replied:
Fixed

zack-rma reviewed (commented):
A few notes to improve the readability and clarity.

zack-rma commented on README.build-gradle.md:
```diff
@@ -0,0 +1,146 @@
+
+![](./README_images/intellij_remote_debug.png)
+
+Update the /wrims-gui/build/installer/WRIMS2_GUI_Start.bat file to include the remote debug vmargs like this:
```
The path `/wrims-gui/build/installer/WRIMS2_GUI_Start.bat` is missing the folder above the `.bat` file. It should be of the format `/wrims-gui/build/installer/wrims_gui_x64_[version_date]/WRIMS2_GUI_Start.bat`

jmdegeorge replied:
Fixed

jmdegeorge committed:
Fixed build.gradle files to support eclipse gradle build. Updated README with PR fix requests.

jmdegeorge committed:
Adding Eclipse build readme file and associated images.
Updated wrims build tasks. Added runWrims and debugWrims tasks. Added WRIMS2_GUI_Debug.bat

adamkorynta commented on README.md:
recommend adding a link to the build-eclipse readme file as well

jmdegeorge replied:
Added.

zack-rma commented on README.build-eclipse.md:
````diff
@@ -0,0 +1,166 @@
+``` 
+
+## 1. Pull Source from GitHub & build the project
+Clone the repository to your local machine.
````
It would be nice to represent these as bullet points, or at least put in line breaks to maintain readability.

jmdegeorge replied:
Fixed

zack-rma commented on README.build-eclipse.md:
````diff
@@ -0,0 +1,166 @@
+```
+git clone https://github.com/CentralValleyModeling/wrims.git
+git switch Feature/wrims-devops
+cd wrims
````
The directory change will need to happen after the clone, before the branch switch.

jmdegeorge replied:
Fixed

zack-rma commented on README.build-eclipse.md:
```diff
@@ -0,0 +1,166 @@
+git clone https://github.com/CentralValleyModeling/wrims.git
+git switch Feature/wrims-devops
+cd wrims
+gradlew.bat build
```
These Gradle commands will need to be prepended with `.\` in a Windows environment. Might be worth making a note of this.

zack-rma replied:
In order to get the Gradle build to run, I needed to set the Java home in the `<USER-DIR>\.gradle\gradle.properties` file. I did this by adding the line: `org.gradle.java.home=C:\\Program Files\\Java\\jdk1.8.0_[version]`. Of course, this will vary per install/Java version.

I'm sure there's a better way to do this on a non-global level, but other methods didn't seem to work for me.

jmdegeorge replied:
Resolved

rma-tom reviewed (commented):
Noted during build process tests: The zipWrimsGui task fails if the wrims-gui/build/installer folder isn't empty before the task starts.

zack-rma commented on README.build-eclipse.md:
```diff
@@ -0,0 +1,166 @@
+ 
+![](./README_images/close_eclipse_project.png)
+
+## 5. Configure Eclipse JRE
```
In my run through, I had to make sure to set the Compiler compliance level to 1.8, as it defaults to 21. This is located in Window->Java->Compiler. It may be helpful to add a not about that.

jmdegeorge committed:
Additional eclipse process corrections and screenshots.

jmdegeorge committed:
Updating eclipse build process screenshots and remote debug instructions.

jmdegeorge committed:
Screenshot fixes and added link to eclipse instructions from root README.md.

jmdegeorge committed:
Cleaning up adjusted process instructions.

jmdegeorge committed:
Removing WRIMS_Debug.bat file from default install build.

jmdegeorge committed:
Refactoring wrims-gui to wrims-install package. Adding eclipse readme updates with notes on how to push code changes into wrims-install.

jmdegeorge committed:
Pushing final review changes to docs and added direct debugWrims tasks for easy launching with remote debugging enabled.

rma-tom reviewed (approved):
Quick pass through to get the updates out where they can be used
